### PR TITLE
Document how gauges convert types to float

### PR DESCRIPTION
**Pull request by jabdoa2**
**What this PR does / why we need it**:
Documentation for #1963

### CODE REVIEW

File: docs/customresourcestate-metrics.md [hunk 1]
```diff
@@ -242,6 +242,60 @@ Produces the metric:
...
+
+```yaml
+status:
+  conditions:
+    - lastTransitionTime: "2019-10-22T16:29:31Z"
+      status: "True"
+      type: Ready
+```
+
+
```
Comment (mrueg) [inline docs/customresourcestate-metrics.md:298]: Can you add the resulting metric here?

**Comment by dgrisonnet** [comment]
Lgtm pending @mrueg comment in https://github.com/kubernetes/kube-state-metrics/pull/1964#discussion_r1095711504

File: docs/customresourcestate-metrics.md (hunk 1 above)
Comment (mrueg) [inline docs/customresourcestate-metrics.md:298]:
```suggestion
kube_customresource_foo_status{customresource_group="myteam.io", customresource_kind="Foo", customresource_version="v1", type="Ready"} 1.571761771e+09
```

**Comment by mrueg** [comment]
I amended the PR with the missing metric in the doc.

File: docs/customresourcestate-metrics.md [hunk 2]
```diff
@@ -242,6 +242,61 @@ Produces the metric:
...
+```yaml
+status:
+  conditions:
+    - lastTransitionTime: "2019-10-22T16:29:31Z"
+      status: "True"
+      type: Ready
+```
+
+kube_customresource_foo_status{customresource_group="myteam.io", customresource_kind="Foo", customresource_version="v1", type="Ready"} 1.571761771e+09
```
Comment (jabdoa2) [inline docs/customresourcestate-metrics.md:298]: Value should be either 1.0 or 0.0 herr

Comment (mrueg) [reply docs/customresourcestate-metrics.md:298]: Whops, you are correct! Thanks, will fix!

Comment (mrueg) [inline docs/customresourcestate-metrics.md:298]:
```suggestion
kube_customresource_foo_status{customresource_group="myteam.io", customresource_kind="Foo", customresource_version="v1", type="Ready"} 1.0
```

**Review by dgrisonnet — APPROVED** [review]
/lgtm
/approve

**Pull request merged**